pkg/provider/aws/plugin.go

```diff
@@ -74,23 +74,54 @@ func (a *AWS) copyConstructEdgeToDag(
 	dep graph.Edge[core.Construct],
 	dag *core.ResourceGraph) error {
 	data := knowledgebase.EdgeData{AppName: a.Config.AppName, Source: sourceResource, Destination: destinationResource}
+
 	switch construct := dep.Source.(type) {
 	case *core.ExecutionUnit:
-		switch dep.Destination.(type) {
+		if a.Config.GetExecutionUnit(construct.ID).Type == kubernetes.KubernetesType {
+			data.SourceRef = construct.AnnotationKey
+		}
+		for _, envVar := range construct.EnvironmentVariables {
+			if envVar.Construct != nil && envVar.Construct.Id() == dep.Destination.Id() {
+				data.EnvironmentVariables = append(data.EnvironmentVariables, envVar)
+			}
+		}
+		switch dest := dep.Destination.(type) {
 		case *core.Orm:
 			data.Constraint = knowledgebase.EdgeConstraint{
 				NodeMustExist: []core.Resource{&resources.RdsProxy{}},
 			}
+			// Because we dont understand resource types, our edges cant depict what is truly valid and what isnt yet. Example
+			// 	found multiple paths which satisfy constraints for edge aws:lambda_function:lambda -> aws:rds_instance:rds.
+			// Paths: [[{*resources.LambdaFunction *resources.RdsProxy} {*resources.RdsProxyTargetGroup *resources.RdsProxy} {*resources.RdsProxyTargetGroup *resources.RdsInstance}]
+			// [{*resources.LambdaFunction *resources.RdsProxy} {*kubernetes.HelmChart *resources.RdsProxy} {*kubernetes.HelmChart *resources.RdsInstance}]]
+			// When we can classify lambda and helm as compute then we can understand that there is no need for the data flow in this manner
```

Comment:
Yeah, this (and the constraint in Kv) is awkward. Do we have a ticket/task for designing the fix for it? Until then, even time we add an edge, it could have unintended consequences by adding a new path between otherwise unrelated things.

Reply:
this should fall into the design task for high level intents that yuval is working on, if not ill cut one and have some ideas about what we would need to do here

Reply:
Maybe down the line, but this problem is more knowledge-base implementation specific that I'm not confident will be covered by that.